questions/qa-bidi-unicode-controls.en.html

```diff
@@ -188,44 +191,98 @@ <h2>Changing the base direction</h2>
+    <p>To achieve the correct effect we add the two invisible control characters,  <span class="uname" style="color: green;">U+2067 RIGHT-TO-LEFT ISOLATE</span> (RLI), and <span class="uname" style="color: red;">U+2069 POP DIRECTIONAL ISOLATE</span> (PDI), represented in the code snippet below as numeric character entities: </p>
   <figure class="example">
       <blockquote><code>
-        <bdo dir="ltr">title=&quot;'&amp;<span style="color: green;" title="RLE">#x202B;</span>...&amp;<span style="color: red;" title="PDF">#x202C;</span>' [in Persian]&quot;</bdo>
+        <bdo dir="ltr">title=&quot;'&amp;<span style="color: green;" title="RLE">#x2067;</span>FREQUENTLY ASKED QUESTIONS&amp;<span style="color: red;" title="PDF">#x2069;</span>' [in persian]&quot;</bdo>
         </code></blockquote>
     </figure>
+
+<div id="out_tooltip" class="browser_output">
+<p class="browser_output_title">Output in your browser (mouse over for result):</p>
+<div class="browser_output_example" lang="ja">
+<p>See <a title="'&#x202B;سوالات متداول (FAQ)&#x202C;' [in Persian]" href="http://www.webstandards.org/learn/faq/faq-farsi-translation/"> another example</a>.</p>
+</div>
+</div>
 </section>
   
 <section id="wrapping">
 <h2>Tightly wrapping opposite-direction phrases</h2>
     <p>In some cases the bidi algorithm copes fine with bidirectional text, and in others it needs some help. In <a href="/International/articles/inline-bidi-markup/"><cite>Inline markup and bidirectional text in HTML</cite></a><cite> </cite>we make the case that the easiest approach to marking up bidirectional text is to put markup at the start and end of each directional change in the text. This doesn't do any harm, it avoids the likelihood of missing a situation where markup is needed, and it makes the life of the content author much simpler.</p>
     <p>A similar recommendation is valid when dealing with Unicode control characters.</p>
-    <p>Another thing to bear in mind is that ranges need to be nested appropriately. If you have an embedded LTR range in a RTL context, and that LTR range has some RTL text inside it, it won't produce the right result if your ranges are side by side rather than nested. Note how the direction changes are embedded in the following example, rather than side by side.</p>
-    <figure class="example">
-    <p><code>the title is <span style="color:red;">&amp;#x202B;</span>AN INTRODUCTION TO <span style="color:red;">&amp;#x202A;</span>c++<span style="color:red;">&amp;pdf;</span><span style="color:red;">&amp;pdf;</span> in arabic.</code></p>
-    </figure>
+    <p>However, it's important to bear in mind is that ranges need to be nested appropriately. If you have an embedded LTR range in a RTL context, and that LTR range has some RTL text inside it, it won't produce the right result if your ranges are side by side rather than nested. Note how the direction changes are embedded in the following example, rather than side by side.</p>
```

Comment:
The extra `is` reads funny. Also would suggest changing `side by side` => `side-by-side`

```suggestion
    <p>However, it's important to bear in mind that ranges need to be nested appropriately. If you have an embedded LTR range in a RTL context, and that LTR range has some RTL text inside it, it won't produce the right result if your ranges are side-by-side rather than nested. Note how the direction changes are embedded in the following example, rather than side-by-side.</p>
```

Reply:
Richard, as a very high level comment, this reads as if it is addressed to someone who is already quite familiar with a large fraction of the issues and for whom you are just filling in a few details.   Depending on your expectations of the audience, that might be ok although I think it should be made more explicit.  The article "[Inline markup and bidirectional text in HTML](https://deploy-preview-449--i18n-drafts.netlify.app/International/articles/inline-bidi-markup/)." might help with that, but the link gets a 404.

The situation is further complicated by some of the examples, which either don't work or leave me thoroughly confused.  Take, e.g., the set that conclude the "Changing the base direction" section.   The proposed fix,
   title="'&#x2067;FREQUENTLY ASKED QUESTIONS&#x2069;' [in persian]"
Says "FREQUENTLY ASKED QUESTIONS" and not "(FAQ)" as shown above, and does not show the string in Persian at all.   Or, if the string "FREQUENTLY ASKED QUESTIONS" is intended to substitute for the Persian, "(FAQ)" is missing.   I can deduce what was probably intended, but I suspect only because I knew what was going on before trying to read the text.

A few paragraphs further up, the reader is told "For this reason, in an ideal world you would want to follow the recommendation of the Unicode Standard to use RLI and...".  If I am to follow those recommendations, I would want to know where I can read that actual text, but no such citation appears to be available.

The first paragraph under "Tightly wrapping..." points me to "[Inline markup and bidirectional text in HTML]: (https://deploy-preview-449--i18n-drafts.netlify.app/International/articles/inline-bidi-markup/)  but that appears to be the same dead link as the one mentioned earlier.  

In addition, unless one has a rather good understanding of what is going on and what interactions might occur, a sentence like "A similar recommendation is valid when dealing with Unicode control characters." is fairly close to meaningless.

Then, under "Dealing with spillover ...", there appears to be an abrupt  and unexplained switch from Arabic to Hebrew and then back to Arabic.

Dare I ask why, if the start and end point pairs are generally preferred to the use of RLM and LRM (the text seems to imply that but does not quite say it), HTML provides named entities for those two but not for RLI, PDI, etc.?

Will the reader know what you mean by a "neutral character"?  And that "neutrals" refers to the same thing?

The tutorial, [Creating HTML Pages in Arabic, Hebrew and Other Right-to-left Scripts](https://deploy-preview-449--i18n-drafts.netlify.app/International/tutorials/bidi-xhtml/) is another dead link as are all of the other links under "Further reading".

The draft is probably in need of another pass.

Reply:
> Dare I ask why, if the start and end point pairs are generally preferred to the use of RLM and LRM (the text seems to imply that but does not quite say it), HTML provides named entities for those two but not for RLI, PDI, etc.?

It's not for want of asking, but the HTML5 editors are against it.

Reply:
All points addressed.

Reply:
> > Dare I ask why, if the start and end point pairs are generally preferred to the use of RLM and LRM (the text seems to imply that but does not quite say it), HTML provides named entities for those two but not for RLI, PDI, etc.?
> 
> It's not for want of asking, but the HTML5 editors are against it.

This sounds like something we should revisit with them.